recipes/pdal/all/conanfile.py

```diff
@@ -33,248 +36,322 @@ class PdalConan(ConanFile):
+        self.cpp_info.components["pdal_base"].set_property("cmake_target_name", "pdalcpp")
+        self.cpp_info.components["pdal_base"].libs = ["pdalcpp"]
         if not self.options.shared:
-            self.cpp_info.components["pdal_base"].libs.extend(["pdal_arbiter", "pdal_kazhdan"])
-            if self.settings.os == "Windows":
-                # dependency of pdal_arbiter
-                self.cpp_info.components["pdal_base"].system_libs.append("shlwapi")
+            self.cpp_info.components["pdal_base"].libs.extend(["pdal_kazhdan", "pdal_lepcc"])
+        self.cpp_info.components["pdal_base"].builddirs.append(os.path.join("lib", "cmake"))
         self.cpp_info.components["pdal_base"].requires = [
-            "pdal_util", "eigen::eigen", "gdal::gdal", "libcurl::libcurl",
-            "libgeotiff::libgeotiff", "nanoflann::nanoflann"
+            "pdal_util",
+            "arbiter::arbiter",
+            "eigen::eigen",
+            "gdal::gdal",
+            "json-schema-validator::json-schema-validator",
+            "libgeotiff::libgeotiff",
+            "nanoflann::nanoflann",
+            "nlohmann_json::nlohmann_json",
+            "proj::proj",
+            "utfcpp::utfcpp",
         ]
         if self.options.with_xml:
             self.cpp_info.components["pdal_base"].requires.append("libxml2::libxml2")
         if self.options.with_zstd:
             self.cpp_info.components["pdal_base"].requires.append("zstd::zstd")
-        if self.options.with_laszip:
-            self.cpp_info.components["pdal_base"].requires.append("laszip::laszip")
         if self.options.with_zlib:
             self.cpp_info.components["pdal_base"].requires.append("zlib::zlib")
         if self.options.with_lzma:
             self.cpp_info.components["pdal_base"].requires.append("xz_utils::xz_utils")
 
         # pdal_util
         self.cpp_info.components["pdal_util"].set_property("cmake_target_name", "pdal_util")
-        self.cpp_info.components["pdal_util"].libs = ["pdal_util"]
         if not self.options.shared:
-            if self.settings.os in ["Linux", "FreeBSD"]:
-                self.cpp_info.components["pdal_util"].system_libs.extend(["dl", "m"])
-        self.cpp_info.components["pdal_util"].requires = ["boost::filesystem"]
+            self.cpp_info.components["pdal_util"].libs = ["pdal_util"]
+        self.cpp_info.components["pdal_util"].builddirs.append(os.path.join("lib", "cmake"))
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.components["pdal_util"].system_libs.extend(["dl", "m", "pthread"])
+        self.cpp_info.components["pdal_util"].requires = [
+            "boost::filesystem",
+            "nlohmann_json::nlohmann_json",
+            "utfcpp::utfcpp",
+        ]
         if self.options.get_safe("with_unwind"):
             self.cpp_info.components["pdal_util"].requires.append("libunwind::libunwind")
 
         # TODO: to remove in conan v2 once cmake_find_package* generators removed
```

Comment:
I think this https://github.com/conan-io/conan-center-index/pull/21296/files#r1485562557 is caused due to this section, but i think you have a better idea on what is going on than i do

Reply:
> I think this https://github.com/conan-io/conan-center-index/pull/21296/files#r1485562557 is caused due to this section, but i think you have a better idea on what is going on than i do

Alright so just a suggestion, but could we simply bump the version to 2.6.3 as in the recent releases PDAL dumped the pdal_util lib. I have something more or less working based on arbiter (https://github.com/conan-io/conan-center-index/pull/21271 but the bumped up version of nlohmann_json)  and gdal (https://github.com/conan-io/conan-center-index/pull/19298) here are the patches:

1. `recipes/pdal/all/conandata.yml`
```
diff --git a/recipes/pdal/all/conandata.yml b/recipes/pdal/all/conandata.yml
index 230802452..642ac3d6d 100644
--- a/recipes/pdal/all/conandata.yml
+++ b/recipes/pdal/all/conandata.yml
@@ -1,9 +1,9 @@
 sources:
-  "2.6.0":
-    url: "https://github.com/PDAL/PDAL/releases/download/2.6.0/PDAL-2.6.0-src.tar.bz2"
-    sha256: "ec50e60544673a72f4300d90df1e70f1cd28055780b789319185590b63e3f688"
+  "2.6.3":
+    url: "https://github.com/PDAL/PDAL/releases/download/2.6.3/PDAL-2.6.3-src.tar.bz2"
+    sha256: "e4d90a3ce4c9681cd3522ca29e73a88ff3b3c713f918693ad03932a6b7680460"
 patches:
-  "2.6.0":
+  "2.6.3":
     - patch_file: "patches/2.6.0-001-nanoflann-1.5.patch"
       patch_type: "portability"
       patch_description: "Add support for nanoflann v1.5.0"
```

2. `recipes/pdal/all/conanfile.py`
```
diff --git a/recipes/pdal/all/conanfile.py b/recipes/pdal/all/conanfile.py
index 9f2effda1..e87ddf84c 100644
--- a/recipes/pdal/all/conanfile.py
+++ b/recipes/pdal/all/conanfile.py
@@ -73,15 +73,15 @@ class PdalConan(ConanFile):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("arbiter/cci.20231005", transitive_headers=True, transitive_libs=True)
+        self.requires("arbiter/cci.20231122", transitive_headers=True, transitive_libs=True)
         self.requires("boost/1.83.0")
         self.requires("eigen/3.4.0", transitive_headers=True, transitive_libs=True)
-        self.requires("gdal/3.8.0", transitive_headers=True, transitive_libs=True)
-        self.requires("json-schema-validator/2.2.0")
+        self.requires("gdal/3.8.3", transitive_headers=True, transitive_libs=True)
+        self.requires("json-schema-validator/2.3.0")
         self.requires("libgeotiff/1.7.1")
         self.requires("nanoflann/1.5.0", transitive_headers=True, transitive_libs=True)
-        self.requires("nlohmann_json/3.11.2", transitive_headers=True, transitive_libs=True)
-        self.requires("proj/9.3.0", transitive_headers=True, transitive_libs=True)
+        self.requires("nlohmann_json/3.11.3", transitive_headers=True, transitive_libs=True)
+        self.requires("proj/9.3.1", transitive_headers=True, transitive_libs=True)
         self.requires("utfcpp/4.0.1")
         if self.options.with_xml:
             self.requires("libxml2/2.11.5", transitive_headers=True, transitive_libs=True)
@@ -183,7 +183,6 @@ class PdalConan(ConanFile):
     def _patch_sources(self):
         apply_conandata_patches(self)
         top_cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
-        util_cmakelists = os.path.join(self.source_folder, "pdal", "util", "CMakeLists.txt")
 
         # Provide these dependencies via the CMakeLists.txt in the recipe instead
         for cmake_module in [
@@ -230,9 +229,6 @@ class PdalConan(ConanFile):
         # Disabling libxml2 support is only possible via patching
         if not self.options.with_xml:
             replace_in_file(self, top_cmakelists, "include(${PDAL_CMAKE_DIR}/libxml2.cmake)", "")
-        # Disabling libunwind support is only possible via patching
-        if not self.options.get_safe("with_unwind", False):
-            replace_in_file(self, util_cmakelists, "include(${PDAL_CMAKE_DIR}/unwind.cmake)", "")
         # Disable rpath manipulation
         replace_in_file(self, top_cmakelists, "include(${PDAL_CMAKE_DIR}/rpath.cmake)", "")
         # Disable copying of symbols from libpdal_util.dylib to libpdalcpp.dylib
@@ -267,13 +263,13 @@ class PdalConan(ConanFile):
         # TODO: to remove in conan v2 once cmake_find_package* generators removed
         self._create_cmake_module_alias_targets(
             os.path.join(self.package_folder, self._module_target_file),
-            {"pdalcpp": "PDAL::pdalcpp", "pdal_util": "PDAL::pdal_util"},
+            {"pdalcpp": "PDAL::pdalcpp"},
         )
 
     def _create_cmake_module_variables(self, module_file):
         pdal_version = Version(self.version)
         content = textwrap.dedent(f"""\
-            set(PDAL_LIBRARIES pdalcpp pdal_util)
+            set(PDAL_LIBRARIES pdalcpp)
             set(PDAL_VERSION_MAJOR {pdal_version.major})
             set(PDAL_VERSION_MINOR {pdal_version.minor})
             set(PDAL_VERSION_PATCH {pdal_version.patch})
@@ -312,7 +308,7 @@ class PdalConan(ConanFile):
             self.cpp_info.components["pdal_base"].libs.extend(["pdal_kazhdan", "pdal_lepcc"])
         self.cpp_info.components["pdal_base"].builddirs.append(os.path.join("lib", "cmake"))
         self.cpp_info.components["pdal_base"].requires = [
-            "pdal_util",
+            "boost::filesystem",
             "arbiter::arbiter",
             "eigen::eigen",
             "gdal::gdal",
@@ -332,21 +328,6 @@ class PdalConan(ConanFile):
         if self.options.with_lzma:
             self.cpp_info.components["pdal_base"].requires.append("xz_utils::xz_utils")
 
-        # pdal_util
-        self.cpp_info.components["pdal_util"].set_property("cmake_target_name", "pdal_util")
-        if not self.options.shared:
-            self.cpp_info.components["pdal_util"].libs = ["pdal_util"]
-        self.cpp_info.components["pdal_util"].builddirs.append(os.path.join("lib", "cmake"))
-        if self.settings.os in ["Linux", "FreeBSD"]:
-            self.cpp_info.components["pdal_util"].system_libs.extend(["dl", "m", "pthread"])
-        self.cpp_info.components["pdal_util"].requires = [
-            "boost::filesystem",
-            "nlohmann_json::nlohmann_json",
-            "utfcpp::utfcpp",
-        ]
-        if self.options.get_safe("with_unwind"):
-            self.cpp_info.components["pdal_util"].requires.append("libunwind::libunwind")
-
         # TODO: to remove in conan v2 once cmake_find_package* generators removed
         self.cpp_info.names["cmake_find_package"] = "PDAL"
         self.cpp_info.names["cmake_find_package_multi"] = "PDAL"

```
Please let me know what you think, hopefully all these PR get merged soon as it would greatly simply the way our code deals with these packages.